software/contrib/menu.py

```diff
@@ -1,5 +1,11 @@
 """See menu.md for details."""
-from europi import bootsplash
+
+from europi import bootsplash, Pin
+
+if Pin(24).value() == 1:
```

Comment:
[required] We should add a comment here as to _why_ we are sleeping, as it is not clear in this code. Also might as well indicate what pin 24 is, if that's not naturally clear from the comment.

Also, since this is a kind if a hack to work around an issue that we don't fully understand, we should identify the conditions for its removal. Otherwise we'll likely be stuck with this code and it's workaround for a long time. I would put these conditions either here in the comment, or in an GitHub issue and record the issue number in the comments. (My preference would be for the latter).

By 'conditions for removal' I mean what would need to change for this code to be removed. I would guess that it would be a new version of micropython that doesn't not exhibit this behavior. This will probably result in us having to document which micropython versions we support, so that we can remove this code once the 'bad' version is unsupported.

Reply:
Yup, I completely agree, I'll add a GitHub issue to track it and reference that in the code comment.

I imagine the condition for removal would simply be a release of MicroPython that fixes it, at which point we can remove the workaround fix an make it clear that previous versions aren't supported